File: Array-Problems/Find-Equilibrium/FindEquilibrium.java
import java.util.Arrays;

public class FindEquilibrium
{
    public static int findEquilibrium(int[] arr)
    {
        int leftSum = 0;
        int rightSum = 0;
        for (int i = 1; i < arr.length; i++)
            rightSum += arr[i];
        for (int i = 0; i < arr.length; i++)
        {
            // ASSERT:
            //      leftSum = arr[0] + ... + arr[i - 1]
            //     rightSum = arr[i + 1] + ... + arr[n - 1]
            if (leftSum == rightSum)
                return i;
            if (i < arr.length - 1)
            {
                leftSum += arr[i];
                rightSum -= arr[i + 1];
            }
        }
        return -1;
    }

    public static void main(String[] args)
    {
        int[][] test_arrs = {
                {},
                {-7, 1, 5, 2, -4, 3, 0},
                {1, 2, 3, 4, 5},
                {3, 1, 2, 9, 4, 2},
                {9, 2, -3, 1},
                {7, 6, -5, -8, 9}
        };
        for (int i = 0; i < test_arrs.length; i++)
        {
            System.out.printf("Test Case #%d: ", i + 1);
            System.out.println(Arrays.toString(test_arrs[i]));
            int index = findEquilibrium(test_arrs[i]);
            System.out.printf("    Equilibrium Index: %d\n", index);
        }
    }
}

File: Array-Problems/Find-Equilibrium/Find-Equilibrium.cs
using System;

public class FindEquilibriumSolution
{
    public static int FindEquilibrium(int[] arr)
    {
        int leftSum = 0;
        int rightSum = 0;
        for (int i = 1; i < arr.Length; i++)
            rightSum += arr[i];
        for (int i = 0; i < arr.Length; i++)
        {
            if (leftSum == rightSum)
                return i;
            if (i + 1 < arr.Length)
            {
                leftSum += arr[i];
                rightSum -= arr[i + 1];
            }
        }
        return -1;
    }

    /// <summary>
    /// The entry point.
    /// </summary>
    public static void Main()
    {
        var testArrs = new[] 
        {
            new int[] {},
            new[] {-7, 1, 5, 2, -4, 3, 0},
            new[] {1, 2, 3, 4, 5},
            new[] {3, 1, 2, 9, 4, 2},
            new[] {9, 2, -3, 1},
            new[] {7, 6, -5, -8, 9}
        };
        for (var i = 0; i < testArrs.Length; i++)
        {
            Console.WriteLine("Test Case #{0}: [{1}]", i + 1, string.Join(", ", testArrs[i]));
            Console.WriteLine("    Equilibrium Index: {0}", FindEquilibrium(testArrs[i]));
        }
    }
}